### PR TITLE
Add hover tooltips to coverage viz.

### PR DESCRIPTION
**Pull request by MarkAZhang**
Updated mock data to reflect the data produced from the pipeline step.

Mock data now contains additional metrics for contigs, reads, coverage bins, and the accession itself.
Mock data now aggregates contigs and reads into hit groups.
Mock data now treats coverage as a sparse array. (the data is a lot smaller now)

Coverage viz now has tooltips for coverage bins and hit groups.

Histogram and GenomeViz now support hover tooltips.

Tooltips will display within a buffer distance, since bars can be very small.

coverage

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix a feature that knowingly causes existing functionality to not work as expected)

# Test and Reproduce
Outline the steps to test or reproduce the PR here.

1. Verified that coverage viz displays without crashing.
2. Verified that non-admin users are not affected.

### CODE REVIEW

File: app/assets/src/components/common/CoverageVizBottomSidebar/CoverageVizBottomSidebar.jsx [hunk 1]
```diff
@@ -234,15 +272,51 @@ export default class CoverageVizBottomSidebar extends React.Component {
...
+        <div className={cs.headerControls}>
+          <Dropdown
+            options={this.getAccessionOptions()}
+            value={get("id", currentAccessionSummary)}
+            label="Accession"
+            onChange={this.setCurrentAccession}
+            rounded
+          />
+          <div className={cs.vizLinkContainer}>
+            <a
```
Comment (tfrcarvalho) [inline app/assets/src/components/common/CoverageVizBottomSidebar/CoverageVizBottomSidebar.jsx]: I have seen a few times... Should we create a link object to guarantee this is done correctly?

File: app/assets/src/components/visualizations/Histogram.js [hunk 2]
```diff
@@ -271,6 +336,18 @@ export default class Histogram {
       }
     }
 
+    this.midpointsToDataIndices = midpointsToDataIndices;
+    this.sortedMidpoints = midpoints.sort((a, b) => a - b);
+
+    this.svg
```
Comment (tfrcarvalho) [inline app/assets/src/components/visualizations/Histogram.js]: Are you doing that refactor soon? I see we perpetuating this appending without data binding issue...

File: app/assets/src/components/visualizations/Histogram.js [hunk 3]
```diff
@@ -173,6 +176,57 @@ export default class Histogram {
     return this.options.barOpacity || 0.8;
   };
 
+  // Find the bar midpoint that is closest to hoverX, within hoverBuffer.
```
Comment (tfrcarvalho) [inline app/assets/src/components/visualizations/Histogram.js]: Can you clarify what a "bar midpoint" is ?
This is not something that applies to all histograms, right?

File: app/assets/src/components/common/CoverageVizBottomSidebar/CoverageVizBottomSidebar.jsx [hunk 4]
```diff
@@ -137,21 +145,47 @@ export default class CoverageVizBottomSidebar extends React.Component {
...
-    }));
+  handleHistogramHover = (pageX, pageY, hoverData) => {
+    const { currentAccessionData } = this.state;
+
+    if (hoverData && hoverData[0] === 0) {
+      this.setState({
+        histogramTooltipParams: {
+          left: pageX,
+          top: pageY,
+          tooltipData: getHistogramTooltipData(
```
Comment (tfrcarvalho) [inline app/assets/src/components/common/CoverageVizBottomSidebar/CoverageVizBottomSidebar.jsx]: I think there is a better way to do this... I would have to understand the context better.
But you typically set state fo the hovered element, and get the data on re-rendering if it changes. 
Definitely, left and top should not be in state.

File: app/assets/src/components/visualizations/Histogram.js (hunk 3 above)
Comment (MarkAZhang) [reply app/assets/src/components/visualizations/Histogram.js]: Changed to "bar x-center". It's the center of the bar on the x-dimension.

File: app/assets/src/components/common/CoverageVizBottomSidebar/CoverageVizBottomSidebar.jsx (hunk 4 above)
Comment (MarkAZhang) [reply app/assets/src/components/common/CoverageVizBottomSidebar/CoverageVizBottomSidebar.jsx]: Refactored into separate "enter", "hover", "exit" events as discussed.

File: app/assets/src/components/common/CoverageVizBottomSidebar/CoverageVizBottomSidebar.jsx [hunk 5]
```diff
@@ -174,23 +230,30 @@ export default class CoverageVizBottomSidebar extends React.Component {
...
+      data.coverage_bin_size
     );
 
     this.contigReadViz = new GenomeViz(
       this.contigReadVizContainer,
       contigReadVizData,
       {
-        domain: [0, data.total_length]
+        domain: [0, data.total_length],
+        colors: ["#A9BDFC", "#3768FA"],
```
Comment (tfrcarvalho) [inline app/assets/src/components/common/CoverageVizBottomSidebar/CoverageVizBottomSidebar.jsx]: Is it worth moving this colors to some kind of color scheme for later use?

**Pull request merged**